### PR TITLE
Refactor: User class methods to convert recipe ID's to full recipes a…

### LINKED ISSUE #107: Fix tests to use imported server data

### LINKED ISSUE #108: Fix user filter and search methods to use imported server data

### PR DESCRIPTION
**Pull request by jfogiato**
Files changed: `User.js`, `test-data.js`, `scripts.js`

Change summary: Updated User search functionality to utilize the new format of data coming from the server. We're getting ID's now, but originally had the class methods set up to receive full `Recipe` class instances. Added a method to convert the ID's to objects, then filter/search. Rewrote tests and test data to accommodate the changes. 

Additional Questions/Concerns: None

Closes #107, Closes #108

### CODE REVIEW

File: src/classes/User.js [hunk 1]
```diff
@@ -5,12 +5,18 @@ class User {
     this.recipesToCook = userData.recipesToCook;
   }
 
-  filterSavedByTag(tag) {
-    return this.recipesToCook.filter(recipe => recipe.tags.includes(tag));
+  convertToFullRecipe(recipeRepo) {
+    return this.recipesToCook.map(userRecipe => {
+      return recipeRepo.recipes.find(recipe => recipe.id === userRecipe)
+    });
```
Comment (qrispi) [inline src/classes/User.js:11]: Love the re-usability of this method.

File: src/data/test-data.js [hunk 2]
```diff
@@ -406,604 +406,17 @@ const users = [
...
-      {
-        "ingredient": 2028,
-        "amount": 3
-      },
-      {
-        "ingredient": 4047,
-        "amount": 2
-      }
-    ]
+    "recipesToCook": []
```
Comment (qrispi) [inline src/data/test-data.js:419]: LOL Joe racking up beaucoup deleted lines again!

File: test/User-test.js [hunk 3]
```diff
@@ -29,63 +29,23 @@ describe('User', () => {
   it('should have an id', () => {
     expect(user.id).to.equal(1);
   });
-    
-  it('should have a pantry', () => {
-    expect(user.pantry).to.be.an('array');
-    expect(user.pantry).to.deep.equal(users[0].pantry);
-  });
-    
+
```
Comment (qrispi) [inline test/User-test.js:32]: Smell ya later Pantry!

File: src/classes/User.js (hunk 1 above)
Comment (jsahim) [reply src/classes/User.js:11]: much easier to follow this code.

**Review by jsahim — APPROVED** [review]
Great work on this code cleanup!

**Review by qrispi — APPROVED** [review]
I still feel like we may have missed a bug here but all tests are passing and the site functions how we expect so ship it!

**Review by LordSchwifty — APPROVED** [review]
Nice work! Love the filterSavedByTag function!

**Pull request merged**